packages/next/src/shared/lib/app-router-context.shared-runtime.ts

```diff
@@ -31,8 +31,7 @@ export type LazyCacheNode = {
    */
   rsc: null
 
-  // TODO: Add prefetchRsc field.
-  // prefetchRsc: null
+  prefetchRsc: React.ReactNode
```

Comment:
This type ended up allowing for non-null values because it could have been cloned from a previous CacheNode entry.

Reply:
added a comment with more explanation